src/core/ext/client_config/lb_policy_factory.h

```diff
@@ -47,8 +47,25 @@ struct grpc_lb_policy_factory {
   const grpc_lb_policy_factory_vtable *vtable;
 };
 
+/** A resolved address alongside any LB related information associated with it.
+ * \a user_data, if not \a NULL, is opaque and meant to be consumed by the gRPC
+ * LB policy. Anywhere else, refer to the functions in \a
+ * grpc_lb_policy_user_data_vtable to operate with it */
+typedef struct grpc_lb_address {
+  grpc_resolved_address *resolved_address;
+  void *user_data;
+} grpc_lb_address;
+
+/** Functions acting upon the opaque \a grpc_lb_address.user_data */
+typedef struct grpc_lb_policy_user_data_vtable {
```

Comment:
It looks like this vtable is only needed because we're requiring the LB policy to do memory management for the user data.  Instead, how about if we make it the caller's responsibility to do that, so that the LB policy only needs to return the right pointer?

In other words, in the gRPC-LB policy, when we pass in the user_data to the RR policy, the user data can be a pointer to the right LB token, which is owned by the gRPC-LB policy.  That way, the RR policy doesn't need to do any copying or destroying; it can simply pass back the pointer to the LB token associated with whichever address it chooses.

I suspect that this approach would both simplify this API and would minimize the code changes needed to the RR code.  What do you think?